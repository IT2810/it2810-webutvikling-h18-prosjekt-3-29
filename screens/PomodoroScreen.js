import React from 'react';
import { ScrollView, StyleSheet, Button, Text, Alert, AsyncStorage} from 'react-native';
import { Countdown } from  '../components/Countdown';

export default class PomodoroScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      finishedPomodoros : '',
      startWork : false,
      startPause : false,
    }
  }

  static navigationOptions = {
    title: 'Pomodoro',
  };

  componentDidMount() {
    this._getFinishedPomodorosFromAsync()
  }

  handleFinished = (finVal) => {
    if (finVal == "work") {
      this.setState({startPause : true})
    } else if (finVal == "pause") {
      this.setState({startPause : false})
      this.setState({startWork : false})
      Alert.alert("Pomodoro finished, gz")
      this._addFinishedPomodoroToAsync();
    }
  }

  toggleButton = () => {
    this.setState({ startWork : !this.state.startWork})
    if (this.state.startPause) {
      this.setState({startPause : false})
    }

  }

  render() {
    return (
        <ScrollView style={styles.container}>
            <Text> Work: </Text> {this.state.startWork ? <Countdown type="work" time="0.1" onWorkFinished ={this.handleFinished}/> : <Text> 25 : 00 : 000</Text>}
            <Text> Pause: </Text> {this.state.startPause ? <Countdown type="pause" time="0.05" onWorkFinished ={this.handleFinished}/> : <Text> 05 : 00 : 000</Text>}
            <Button title={this.state.startWork ? "stop" : "start" } onPress={this.toggleButton}/>
            
            <Text>{"\n\n\n\n"} Finished Pomodoros : {this.state.finishedPomodoros} </Text> 
        </ScrollView>
    );
  }






  //funksjoner for get og set av async-item

  _addFinishedPomodoroToAsync = async () => {

    let counter = String(this.state.finishedPomodoros + 1)

    try {
      await AsyncStorage.setItem('pomodoroCounter', counter);
      this.setState({finishedPomodoros : parseInt(counter)})
    } catch (error) {
      // Error retrieving data
      console.log(error.message);
    }
      
  }

  _getFinishedPomodorosFromAsync = async () => {
    console.log("async")
    try {
      let donePomodoros = await AsyncStorage.getItem('pomodoroCounter');
     if (donePomodoros == null) {
       console.log("fin: " + 0)
      this.setState({finishedPomodoros : 0})
     }
     else {
       donePomodoros = parseInt(donePomodoros)
       this.setState({finishedPomodoros : donePomodoros})
     }
   } catch (error) {
     // Error retrieving data
     console.log(error.message);
   }
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
});
