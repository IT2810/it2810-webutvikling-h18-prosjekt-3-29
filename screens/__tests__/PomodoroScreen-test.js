import React from 'react';
import PomodoroScreen from '../PomodoroScreen';
import renderer from 'react-test-renderer';

it('should render corrrectly', async () => {
    const tree = renderer.create(<PomodoroScreen/>).toJSON();
    expect (tree).toMatchSnapshot();
  })