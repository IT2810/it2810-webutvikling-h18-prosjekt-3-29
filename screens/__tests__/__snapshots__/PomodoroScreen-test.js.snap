// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render corrrectly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "paddingTop": 15,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4b4b4b",
        "fontSize": 30,
      }
    }
  >
      
    idle
      
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#4b4b4b",
          "borderRadius": 150,
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 300,
            "width": 150,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 150,
              "height": 300,
              "transform": Array [
                Object {
                  "translateX": 75,
                },
                Object {
                  "rotate": "0deg",
                },
                Object {
                  "translateX": -75,
                },
              ],
              "width": 150,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 300,
            "width": 150,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "backgroundColor": "#4b4b4b",
              "borderRadius": 150,
              "height": 300,
              "transform": Array [
                Object {
                  "translateX": 75,
                },
                Object {
                  "rotate": "0deg",
                },
                Object {
                  "translateX": -75,
                },
              ],
              "width": 150,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 145,
            "height": 290,
            "width": 290,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4b4b4b",
            "fontSize": 30,
          }
        }
      >
         25 : 00 
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    

  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        start
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4b4b4b",
      }
    }
  >
    


     Finished Pomodoros : 
    
     
  </Text>
</View>
`;
